pallets/xdns/src/lib.rs

```diff
@@ -552,17 +560,23 @@ pub mod pallet {
             asset_id: AssetId,
             target_id: TargetId,
             token_info: TokenInfo,
+            token_location: Option<MultiLocation>,
         ) -> DispatchResultWithPostInfo {
             ensure_root(origin.clone())?;
 
             assert!(!Self::check_asset_is_mintable(target_id, asset_id));
 
             if !<AllTokenIds<T>>::get().contains(&asset_id) {
-                Self::register_new_token(&origin, asset_id, token_info.clone())?;
```

Comment:
would u mind exposing "register_new_gateway" via extrinsic too?

Reply:
Isn't  "register_new_gateway" exposed via extrinsic in `pallet-portal`? or you mean another extrinsic in `pallet-xdns`?

Reply:
added in [#1c37c48](https://github.com/t3rn/t3rn/pull/1469/commits/1c37c4856309b1ed856490631936d1eee630c136)